tests/graceful-recovery.md

```diff
@@ -0,0 +1,97 @@
+5. Follow the [installation instructions](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/docs/installation.md)
+to deploy NGINX Gateway Fabric.
+6. In a separate terminal track NGF logs by running `kubectl -n nginx-gateway logs -f deploy/nginx-gateway`
+7. In a separate terminal track nginx container logs by running
+`kubectl -n nginx-gateway logs -f <NGF_POD> -c nginx`
+8. Exec into the nginx container inside of the NGF pod by running
+`kubectl exec -it -n nginx-gateway <NGF_POD> --container nginx -- bin/sh`
+9. Inside the nginx container, navigate to `/etc/nginx/conf.d` and ensure that
+`http.conf` and `config-version.conf` look correct.
+10. In a different terminal, deploy the
+[https-termination example](https://github.com/nginxinc/nginx-gateway-fabric/tree/main/examples/https-termination).
+11. Inside the nginx container, check `http.conf` and `config-version.config` to see
+if the configuration and version were correctly updated.
+12. Send traffic through the example application and ensure it is working correctly.
+
+## Testing when nginx-gateway container restarts
+
+1. Ensure NGF and nginx container logs are set up and traffic flows through the example application correctly.
+2. Insert ephemeral container in NGF Pod and kill the nginx-gateway process.
+    1. `kubectl debug -it -n nginx-gateway <NGF_POD> --image=busybox:1.28 --target=nginx-gateway`
+    2. run `ps -A`
+    3. run `kill <nginx-gateway_PID>` (Command should start with `/usr/bin/gateway`)
+3. Check for errors in the NGF and nginx-container logs.
+4. When the nginx-gateway container is back up, ensure traffic flows through the example application correctly.
+5. Open up the NGF and nginx container logs and check for errors.
+6. Inside the nginx container, check that `http.conf` was not changed and `config-version.conf` had its version set to `2`.
+7. Send traffic through the example application and ensure it is working correctly.
+8. Check that NGF can still update statuses of resources.
+   1. Delete the HTTPRoute resources by running `kubectl delete -f cafe-routes.yaml` in `/examples/https-termination`
+   2. Inside the terminal which is inside the nginx container, check that `http.conf` and
+   `config-version.conf` were correctly updated.
+   3. Send traffic through the example application using the updated resources and ensure traffic does not flow.
+   4. Apply the HTTPRoute resources by running `kubectl apply -f cafe-routes.yaml` in `/examples/https-termination`
+   5. Inside the terminal which is inside the nginx container, check that `http.conf` and
+   `config-version.conf` were correctly updated.
+   6. Send traffic through the example application using the updated resources and ensure traffic flows correctly.
+
+## Testing when nginx container restarts
+
+1. Ensure NGF and nginx container logs are set up and traffic flows through the example application correctly.
+2. Insert ephemeral container in NGF Pod and kill the nginx-master process.
```

Comment:
Do you need an ephemeral container for this test case or can you just exec into the nginx container and kill the master process?

Reply:
I just double checked and yes I can kill the master process from the nginx container. Do you think that would be simpler?

Reply:
Yeah, I think that would be simpler.